src/goto-unwind/src/goto_unwind_parse_options.cpp

```diff
@@ -0,0 +1,157 @@
+#include <goto-programs/read_goto_binary.h>
+#include <goto-programs/remove_function_pointers.h>
+#include <goto-programs/write_goto_binary.h>
+#include <util/string2int.h>
+#include <util/validate.h>
+#include <util/version.h>
+
+goto_unwind_parse_optionst::goto_unwind_parse_optionst(
+  int argc,
+  const char **argv)
+  : parse_options_baset("(unwind-limit):", argc, argv, "goto-unwind")
+{
+}
+
+struct goto_unwind_optionst
+{
+  std::size_t unwind_limit;
+  std::string source_goto_binary_filepath;
+  std::string destination_goto_binary_filepath;
+
+  static constexpr const char *unwind_limit_option_name = "unwind-limit";
+
+  static std::ostream &help(std::ostream &log)
+  {
+    log << "--" << unwind_limit_option_name << " N    How far to unwind\n";
+    return log;
+  }
+
+  /// Parse an integral-type option. Returns nullopt if the option was not
+  /// passed, returns a filled-in option with the value that was passed if the
+  /// option was passed, throws an exception if the option was passed but the
+  /// option value couldn't be parsed.
+  template <typename T>
+  static optionalt<T> parse_int_option(
+    const cmdlinet &cmdline,
+    const std::string &option_name,
+    const std::string &error_message)
+  {
+    if(cmdline.isset(option_name.c_str()))
+    {
+      if(
+        auto option_value =
+          string2optional<T>(cmdline.get_value(option_name.c_str())))
+      {
+        return option_value;
+      }
+      else
+      {
+        throw invalid_command_line_argument_exceptiont{
+          error_message +
+            "(value was: " + cmdline.get_value(option_name.c_str()) + ")",
+          "--" + option_name};
+      }
+    }
+    return nullopt;
+  }
+
+  static goto_unwind_optionst from_cmdline(const cmdlinet &cmdline)
+  {
+    goto_unwind_optionst goto_unwind_options{};
+
+    if(cmdline.args.size() == 1)
+    {
+      goto_unwind_options.source_goto_binary_filepath =
+        goto_unwind_options.destination_goto_binary_filepath = cmdline.args[0];
+    }
+    else if(cmdline.args.size() == 2)
+    {
+      goto_unwind_options.source_goto_binary_filepath = cmdline.args[0];
+      goto_unwind_options.destination_goto_binary_filepath = cmdline.args[1];
+    }
+    else
+    {
+      throw invalid_command_line_argument_exceptiont{
+        "Expecting exactly one or two arguments",
+        "<Source binary> [<Destination binary>]"};
+    }
+
+    goto_unwind_options.unwind_limit =
+      parse_int_option<std::size_t>(
+        cmdline, unwind_limit_option_name, "couldn't parse as size_t")
+        .value_or(10);
+    return goto_unwind_options;
+  }
+};
+
+int goto_unwind_parse_optionst::doit()
+{
+  const auto options = goto_unwind_optionst::from_cmdline(cmdline);
+  auto goto_model = get_goto_model(options.source_goto_binary_filepath);
+  do_unwind(goto_model, options.unwind_limit);
+  write_goto_binary(
+    options.destination_goto_binary_filepath,
+    goto_model,
+    log.get_message_handler());
+  return 0;
```

Comment:
:no_entry_sign: We should use `CPROVER_EXIT_SUCCESS` instead of this magic value.

Reply:
ok